File: Server/src/models/Account.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IAccount extends Document {
  username: string;
  password_hash: string;
  email: string;
  created_at: Date;
}

const AccountSchema: Schema = new Schema({
  username: { type: String, required: true, unique: true },
  password_hash: { type: String, required: true },
  email: { type: String, required: true, unique: true },
  created_at: { type: Date, default: Date.now }
});

export const AccountModel = mongoose.model<IAccount>('Account', AccountSchema);

File: Server/Models/Account.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace PokemonMMO.Models;

/// <summary>
/// User account — authentication credentials.
/// </summary>
public class Account
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = null!;

    [BsonElement("username")]
    public string Username { get; set; } = null!;

    [BsonElement("password_hash")]
    public string PasswordHash { get; set; } = null!;

    [BsonElement("email")]
    public string Email { get; set; } = null!;

    [BsonElement("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
